### PR TITLE
WIP: Update regular release documentation

### PR DESCRIPTION
**Pull request by garbas**

### CODE REVIEW

File: src/releng_docs/deploy/regular.rst [hunk 1]
```diff
@@ -1,31 +1,73 @@
...
+   projects to **staging channel**.
+
+   To trigger automatic deployment to staging channel you need to force push
+   from ``master`` to ``staging`` branch.
 
    .. code-block:: console
 
         $ git clone git@github.com:mozilla/release-services.git
         $ cd release-services
         $ git push -f origin origin/master:staging
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: Maybe we could remove the `$` prefix to make it easier to copy/paste these blocks?

File: src/releng_docs/deploy/regular.rst [hunk 2]
```diff
@@ -1,31 +1,73 @@
...
+   Once pushed to staging branch a deployment will start via taskcluster github
+   integration. You can find a link to taskcluster deployment group at the
+   `page listing commits of staging branch`_.
+
+   .. image:: page_listing_commits_of_staging_branch.png
+
+   To make everybody aware that this is happening announce staging deployment on
+   `#release-services` IRC channel.::
+
+      I'm pushed commit `<SHORT_SHA_OF_COMMIT>' to staging branch. Until
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: s/I'm/I've/.

Nit: Is it really necessary to copy/paste the commit SHA? I'd just say "I've pushed to staging branch" (people can easily find the SHA if they really need it).

File: src/releng_docs/deploy/regular.rst [hunk 3]
```diff
@@ -1,31 +1,73 @@
...
+      tomorrow, when we deploy to production, staging branch is closed.
+      Deployment to staging is happening here: `<LINK_TO_TASKCLUSTER>'.
+
+   Now you need to close staging branch by checking **Protect this branch** via
+   `staging settings page`_.
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: Maybe move the branch-protection before the announcement? Or rephrase the announcement to say "staging branch will now be closed".

File: src/releng_docs/deploy/regular.rst [hunk 4]
```diff
@@ -1,31 +1,73 @@
...
+
+   .. image:: staging_settings_page.png
+
+#. Testing projects on staging channel
+
+   When deployment succeeds send email to ``release-services@mozilla.com``
+   announcing that staging channel was successful and that maintainers need to
+   test their projects.
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: Why do this via email? I think that going through the list of projects, and pinging the right people on IRC could be enough.

File: src/releng_docs/deploy/regular.rst [hunk 5]
```diff
@@ -1,31 +1,73 @@
...
+   .. todo:: email template,  QA feedback need to be collected via email.
+   .. todo:: how to list maintainers of projects?
+
+
+#. Wednesday meeting
+
+   .. todo:: go or no go decision
+
+      are there any manual things we need to do
+      when do we do a release? since maintainers need to be able to verify
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: This entire section isn't clear ("Wednesday meeting").

File: src/releng_docs/deploy/regular.rst [hunk 6]
```diff
@@ -1,31 +1,73 @@
...
+    
+
 #. Verify that all the production projects in staging that they are functioning
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: "Verify that all production projects are functioning properly in staging."

Also, this section should probably be merged with the one above the "Wednesday meeting" (i.e. "Testing projects on staging channel").

File: src/releng_docs/deploy/regular.rst [hunk 7]
```diff
@@ -1,31 +1,73 @@
...
    properly. Each project should have a list of steps that you can easily
    verify that a deployment was sucessful.
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: TODO: Add a link to all projects: https://docs.mozilla-releng.net/#projects

File: src/releng_docs/deploy/regular.rst (hunk 4 above)
Comment (jankeromnes) [reply src/releng_docs/deploy/regular.rst]: Also, earlier you suggested filing a GitHub issue with checkboxes that everyone can check once their project is verified on staging. I really like this idea (and much prefer it to emails).

File: src/releng_docs/deploy/regular.rst [hunk 8]
```diff
@@ -44,6 +86,8 @@ Protocal that we follow is:
    - announce in ``#ci`` channel that a push to production is about to
      happen.
 
+     TODO: direct this to the person on duty
+
      Example message::
 
          I am about to release a new version of mozilla/release-services
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: s/I am/I'm/

File: src/releng_docs/deploy/regular.rst [hunk 9]
```diff
@@ -65,16 +109,20 @@ Protocal that we follow is:
 #. Push to ``production``. Create a merge commit of master branch and tag it.
    Don't forget to push just created tag.
 
+  TODO: is this the correct branching model
+
    .. code-block:: console
 
        $ git clone git@github.com/mozilla/release-services.git
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: Same as above, maybe remove the `$` prefixes for easier block copy/paste/execute.

File: src/releng_docs/deploy/regular.rst [hunk 10]
```diff
@@ -83,6 +131,7 @@ Protocal that we follow is:
 
    .. todo:: need to explain how to revert when a deployment goes bad.
 
+TODO: we can already do this while waiting for the release to happen
 #. Fill in the release notes on GitHub
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: TODO: Move this entire section above "Verify that all production projects are now deployed", and say "While the production deployment happens, you can fill in the release notes on GitHub".

TODO: Link to https://github.com/mozilla/release-services/releases

File: src/releng_docs/deploy/regular.rst [hunk 11]
```diff
@@ -1,31 +1,73 @@
...
-Protocal that we follow is:
+Release schedule is published in `Release Services calendar`_. Protocol we follow is:
 
+TODO: this should happen on Wed morning, point to the calendar
 
-#. Prior to release a push to ``staging`` branch must happen. This will
-   trigger a deploy of all projects to staging environments.
+#. Push to staging channel.
+
+   A day before pushing to production, on Wednesday morning, we push to
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: Make "day before" bold.

File: src/releng_docs/deploy/regular.rst [hunk 12]
```diff
@@ -44,6 +86,8 @@ Protocal that we follow is:
    - announce in ``#ci`` channel that a push to production is about to
```
Comment (jankeromnes) [inline src/releng_docs/deploy/regular.rst]: Nit: Add "On the **release day** (Thursday)" before "announce that a new deployment to production is going to happen".

File: src/releng_docs/deploy/regular.rst (hunk 3 above)
Comment (La0) [reply src/releng_docs/deploy/regular.rst]: Link to staging protection : https://github.com/mozilla/release-services/settings/branches/staging

File: src/releng_docs/deploy/regular.rst (hunk 8 above)
Comment (La0) [reply src/releng_docs/deploy/regular.rst]: Single line for the IRC message, otherwise it's a PITA to copy/paste

**Comment by La0** [comment]
The base image build does not work anymore if you update the `VERSION` file **before** runing `./please -vv tools base-image`.

I had to revert the change on `VERSION`, then use:

```
./please -vv tools base-image \
    --docker-repo="mozillareleng/services" \
    --docker-tag="base-$(cat ./VERSION)" \
    --docker-username="..." \
    --docker-password="..."
```

**Comment by La0** [comment]
The doc also misses a comment about the mandatory usage of a local nix 2.0 for building the base image (lost 5 minute there, before reminding the way to build)

**Comment by garbas** [comment]
This PR is now obosolete and replaced by #1562

**Pull request closed without merging**